### PR TITLE
Add list_dois{_test}.py

### PR DESCRIPTION
**Pull request by skearnes**
Prints out a markdown-ready list of DOIs and their associated Datasets.

### CODE REVIEW

File: ord_schema/scripts/list_dois.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        dataset_id = os.path.splitext(os.path.basename(filename))[0]
+        assert dataset.dataset_id == dataset_id
+        doi_set = set()
+        for reaction in dataset.reactions:
+            # Some poorly-validated DOI entries start with 'doi:'...
+            match = re.fullmatch(r'(?:(?:doi)|(?:DOI))?:?\s*(.*)',
+                                 reaction.provenance.doi)
+            doi_set.add(match.group(1))
+        if len(doi_set) != 1:
+            raise ValueError(f'Dataset covers multiple DOIs: {doi_set}')
```
Comment (connorcoley) [inline ord_schema/scripts/list_dois.py:57]: Is that an issue worth raising a ValueError for?

File: ord_schema/scripts/list_dois_test.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        dataset = dataset_pb2.Dataset()
+        dataset.dataset_id = 'ord_dataset-1'
+        dataset.reactions.add().provenance.doi = 'foo/bar'
+        dataset.reactions.add().provenance.doi = 'doi:foo/bar'
+        dataset.reactions.add().provenance.doi = 'doi: foo/bar'
+        dataset.reactions.add().provenance.doi = 'DOI:foo/bar'
+        dataset.reactions.add().provenance.doi = 'DOI: \t foo/bar'
+        tempdir = self.create_tempdir()
+        message_helpers.write_message(
+            dataset, os.path.join(tempdir, f'{dataset.dataset_id}.pbtxt'))
```
Comment (connorcoley) [inline ord_schema/scripts/list_dois_test.py:38]: Would you mind changing this test to use .pb files? I expect we'd want to run the script on pb files over pbtxt for speed in parsing.

**Review by connorcoley — APPROVED** [review]

File: ord_schema/scripts/list_dois_test.py (hunk 2 above)
Comment (skearnes) [reply ord_schema/scripts/list_dois_test.py]: Done.

File: ord_schema/scripts/list_dois.py (hunk 1 above)
Comment (skearnes) [reply ord_schema/scripts/list_dois.py]: Updated to list a dataset under each DOI it covers.

**Pull request merged**